docker-storage-setup.sh

```diff
@@ -878,42 +877,50 @@ get_existing_storage_driver() {
   return 1
 }
 
-docker_root_volume_exists() {
-  lvs $VG/$DOCKER_ROOT_LV_NAME > /dev/null 2>&1 && return 0
+extra_volume_exists() {
+  local lv_name=$1
+  lvs $VG/$lv_name > /dev/null 2>&1 && return 0
   return 1
 }
 
-# This returns the mountpoint of $DOCKER_ROOT_LV_NAME
-docker_root_lv_mountpoint() {
+# This returns the mountpoint of $1
+extra_lv_mountpoint() {
   local mounts
-  mounts=$(findmnt -n -o TARGET --first-only --source /dev/$VG/$DOCKER_ROOT_LV_NAME)
+  local lv_name=$1
+  mounts=$(findmnt -n -o TARGET --first-only --source /dev/$VG/$lv_name)
   echo $mounts
 }
 
-# Mount docker root volume on $DOCKER_ROOT_DIR.
-# drop a systemd mount unit configuration file at /etc/systemd/system.
-# This would allow the $DOCKER_ROOT_DIR mountpoint to persist across reboots.
-mount_docker_root_volume() {
+# This method is used to setup mountfile for both, $docker_root_lv_name
+# and $container_root_lv_name. $container_root_lv_name will be mounted
+# on $resolved_mount_dir_path. $docker_root_lv_name
+# will be mounted on $docker_root_dir. Drop a systemd mount unit
+# configuration file at /etc/systemd/system.
+# This would allow the mountpoint to persist across reboots.
+mount_extra_volume() {
   local filename
-  # filename must match $DOCKER_ROOT_DIR path. e.g if $DOCKER_ROOT_DIR is /var/lib/docker
-  # then filename will be var-lib-docker.mount
-  filename=$(systemd_escaped_filename ${DOCKER_ROOT_DIR})
+  local lv_name=$1
+  local mount_dir=$2
+  # filename must match the path ${mount_dir}.
+  # e.g if ${mount_dir} is /var/lib/containers
+  # then filename will be var-lib-containers.mount
+  filename=$(systemd_escaped_filename ${mount_dir})
   cat <<EOF > /etc/systemd/system/$filename
 [Unit]
-Description=Mount docker-root-lv on docker root directory.
+Description=Mount $lv_name on $mount_dir directory.
 Before=docker-storage-setup.service
```

Comment:
I think at some point of time we will have to change it. For now let us leave it as it and take care of it later.